internal/services/subscription/subscription_resource.go

```diff
@@ -92,6 +94,12 @@ func resourceSubscription() *pluginsdk.Resource {
 				},
 			},
 
+			"management_group_id": {
+				Type:        pluginsdk.TypeString,
+				Description: "The Management Group ID to which the subscription belongs",
+				Optional:    true,
+			},
+
```

Comment:
Since this field is present in the old model, I suspect there's a good reason we didn't include it in this resource. There is also the `azurerm_management_group_subscription_association` resource and the `subscription_ids` property on `azurerm_management_group`, so perhaps we should leave this out for now?

(If we _do_ keep it, we'll need some additional test coverage and docs)